src/diffusers/workflow_utils.py

```diff
@@ -0,0 +1,91 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module for managing workflows."""
+import os
+from typing import Union
+
+from .configuration_utils import ConfigMixin
+from .utils import PushToHubMixin, logging
+from .utils.constants import WORKFLOW_NAME
+
+
+logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
+
+_NON_CALL_ARGUMENTS = {"_name_or_path", "scheduler_config", "_class_name", "_diffusers_version"}
+
+
+class Workflow(dict, ConfigMixin, PushToHubMixin):
```

Comment:
```suggestion
class Workflow(dict, PushToHubMixin):
```

Do we really need the ConfigMixin here, the more light-weight the better IMO

Reply:
The reason it's there is because we can easily use `save_config()` (which is ultimately called from `save_workfllow()`). I can definitely copy over the functions being necessary to have a robust `Workflow` class but wanted to be able to reuse things here. But I see your point. Maybe having `save_workflow()` explicitly written here is better.